### DISCUSSION TITLE
How to interactively debug a script or test of an request/collection?

### DISCUSSION THREAD

**Discussion by tho-gru-38** [usebruno/bruno#2319] [Q&A] [answered]
Hi,

As I am new to JavaScript I want to go through my scripts step by step in a debugger.

I know the console.log methods but it is time consuming to adapt the calls an rerun the request.

Therefore I tried the **debugger**-statement but it does not have any viewable effect.

Is there a possibility to start a debugger to analyze the scripts executed scripts of request/collections?

Kind Regards,
Thomas

**Comment by tho-gru-38** [comment]
Hi,

Unfortunately nobody reacts on this question.

How do analyze what is going on in your scripts?
Only console.log() messages? This technique reminds me at writing my first program programming Basic in 1980 on a Apple 2.
Are the better approaches than console.log()?

Kind Regards
Thomas

**Reply by Its-treason** [reply, marked as answer]
Because the script runs on the main process and is isolated with `vm2`, the `debugger` statement does not work. I don't think there is a better way then using `console.log`

**Comment by ant0x64** [comment]
You can use debugger with Bruno CLI
